packages/go_router/lib/src/router.dart

```diff
@@ -29,7 +30,9 @@ import 'typedefs.dart';
 ///
 /// The `redirect` does top-level redirection before the URIs are parsed by
 /// the `routes`. Consider using [GoRoute.redirect] for individual route
-/// redirection.
+/// redirection. If [BuildContext.dependOnInheritedWidgetOfExactType] is used
+/// during the redirection (which is how `of` methods are usually implemented),
+/// a re-evaluation will be triggered when the [InheritedWidget] changes.
```

Comment:
Should this suggest using `SynchronousFuture`?

Reply:
change to futureor